src/packet_analysis/protocol/iptunnel/IPTunnel.cc

```diff
@@ -178,11 +178,16 @@ std::unique_ptr<Packet> build_inner_packet(Packet* outer_pkt, int* encap_index,
 	{
 	auto inner_pkt = std::make_unique<Packet>();
 
+	// Use the original packet's caplen difference to calculate a new caplen for the inner
+	// packet. Previously we just used the len for both values but that's likely not correct
+	// in most cases.
+	uint32_t cap_len = len - (outer_pkt->len - outer_pkt->cap_len);
```

Comment:
Hmm... maybe this is the crux: I think we should instead (or also) update the wire length of the inner packet (and use where appropriate) instead of having `ForwardPacket's `len` be the wire length.

Reply:
See branch linked above. Resolving.